pkg/kapis/devops/v1alpha2/devops.go

```diff
@@ -202,20 +210,129 @@ func (h *ProjectPipelineHandler) GetPipelineRunNodes(req *restful.Request, resp
 	resp.WriteAsJson(res)
 }
 
-func (h *ProjectPipelineHandler) SubmitInputStep(req *restful.Request, resp *restful.Response) {
+func (h *ProjectPipelineHandler) approvableCheck(nodes []clientDevOps.NodesDetail, req *restful.Request) {
+	currentUserName, roleName := h.getCurrentUser(req)
+	// check if current user belong to the admin group, grant it if it's true
+	isAdmin := roleName == iamv1alpha2.PlatformAdmin
+
+	for i, node := range nodes {
+		if node.State != clientDevOps.StatePaused {
+			continue
+		}
+
+		for j, step := range node.Steps {
+			if step.State != clientDevOps.StatePaused || step.Input == nil {
+				continue
+			}
+
+			nodes[i].Steps[j].Approvable = isAdmin || step.Input.Approvable(currentUserName)
+		}
+	}
+}
+
+func (h *ProjectPipelineHandler) createdBy(projectName string, pipelineName string, currentUserName string) bool {
+	if pipeline, err := h.devopsOperator.GetPipelineObj(projectName, pipelineName); err == nil {
+		if creator, ok := pipeline.Annotations[constants.CreatorAnnotationKey]; ok {
+			return creator == currentUserName
+		}
+	} else {
+		log.Error(fmt.Sprintf("cannot get pipeline %s/%s, error %#v", projectName, pipelineName, err))
+	}
+	return false
+}
+
+func (h *ProjectPipelineHandler) getCurrentUser(req *restful.Request) (username, roleName string) {
+	var userInfo user.Info
+	var ok bool
+	var err error
```

Comment:
Please maintain a unified style to declare variables. This Pr uses three methods:

```
// first
var name type
// second
var (
    name type
)
// three
name := function (){ return value}

Reply:
I've changed it. Please see blew:

https://github.com/kubesphere/kubesphere/pull/3156/commits/4d19c4dbb400a4a6c4372777fcaef3f3e3d2a90c#diff-38024724898ee9b3e4ca7f645d4c19869b100c394ccf587ef1e1ea554f3a7a2aR315
https://github.com/kubesphere/kubesphere/pull/3156/commits/4d19c4dbb400a4a6c4372777fcaef3f3e3d2a90c#diff-38024724898ee9b3e4ca7f645d4c19869b100c394ccf587ef1e1ea554f3a7a2aR521

But I believe that different declaration styles have different benefits. I prefer not to restrict it. And I didn't find any documents about this. So please propose a coding style if you think it's very necessary.